testsuite/features/support/twopence_init.rb

```diff
@@ -288,3 +281,51 @@ def file_inject(node, local_file, remote_file)
                   'sle12sp4_terminal'         => $sle12sp4_terminal,
                   'sle15sp3_buildhost'        => $sle15sp3_buildhost,
                   'sle15sp3_terminal'         => $sle15sp3_terminal }
+
+# This is the inverse of `node_by_host`.
+# For each node we choose as key the most meaninful host, i.e. we drop aliases.
+# For instance, we ignore the alias 'ssh_spack_migrated_minion`, we use 'ssh_minion`.
+$host_by_node = {}
+$node_by_host.each do |host, node|
+  next if node.nil?
+  next if node.is_a?(String) && node.empty?
+
+  ignored_hosts = %w[sle_migrated_minion sle_ssh_tunnel_client
+                     sle_spack_migrated_minion sle_ssh_tunnel_minion
+                     ceos_ssh_minion ubuntu_ssh_minion]
+  next if ignored_hosts.include? host
+
+  $host_by_node[node] = host
+  puts "Added #{host} as key for node: '#{node}'"
+end
+
+def client_public_ip(host)
+  node = $node_by_host[host]
+  interface = case host
+              when /^sle/, /^ceos/, /^debian/, 'server', 'proxy', 'build_host'
```

Comment:
Be caution here, I would say it covers all hosts in BV but just be sure about it :)